src/java/org/apache/cassandra/service/StorageService.java

```diff
@@ -888,36 +889,35 @@ private void joinTokenRing(int delay, boolean autoBootstrap, Collection<String>
+            boolean previousDataFound = false;
 
-                if (BOOTSTRAP_DISK_USAGE_THRESHOLD != null && percentageDataDirUtilization > Double.valueOf(BOOTSTRAP_DISK_USAGE_THRESHOLD))
-                {
-                    // disable node preventing bootstrap continuation.
-                    recordNonTransientError(NonTransientError.BOOTSTRAP_ERROR,
-                                            ImmutableMap.of("directory", mostUtilizedDataDir.getKey().location.toString(),
-                                                            "percentageDataDirUtilization", percentageDataDirUtilization.toString(),
-                                                            "threshold", BOOTSTRAP_DISK_USAGE_THRESHOLD.toString()));
-                    logger.error("Preventing node from continuing after failed bootstrap as data_file_dirs are too full ({}%) " +
-                                 "and exceed palantir_cassandra.bootstrap_disk_usage_threshold_percentage ({}%) ",
-                                 percentageDataDirUtilization,
-                                 BOOTSTRAP_DISK_USAGE_THRESHOLD);
-                    unsafeDisableNode();
-                    // leave node in non-transient error state and prevent it from bootstrapping into the cluster
-                    throw new BootstrappingSafetyException(Mode.NON_TRANSIENT_ERROR.toString(),
-                                                          mostUtilizedDataDir.getKey().location.toString(),
-                                                          percentageDataDirUtilization,
-                                                          Double.valueOf(BOOTSTRAP_DISK_USAGE_THRESHOLD));
-                }
-                else
+            for (String keyspaceName : Schema.instance.getNonSystemKeyspaces())
+            {
+                Keyspace keyspace = Keyspace.open(keyspaceName);
+                for (ColumnFamilyStore store : keyspace.getColumnFamilyStores())
                 {
-                    logger.warn("Detected previous bootstrap failure; retrying");
+                    Collection<SSTableReader> tables = store.getSSTables();
+                    if (tables.size() > 0)
+                    {
+                        logger.error("Found previous SSTables {} for keyspace {} and cf {}.", tables, keyspaceName, store.name);
+                        previousDataFound = true;
+                    }
                 }
             }
+
+            if (previousDataFound)
+            {
+                recordNonTransientError(NonTransientError.BOOTSTRAP_ERROR,
+                                        ImmutableMap.of("previousDataFound", "true"));
+                unsafeDisableNode();
+                // leave node in non-transient error state and prevent it from bootstrapping into the cluster
+                throw new BootstrappingSafetyException("Detected data from previous bootstrap, failling.");
+            }
+
+            if (SystemKeyspace.bootstrapInProgress())
```

Comment:
How likely is this to happen? I would imagine that we'd almost immediately create the keyspaces.

Reply:
I just re-added this to match trunk; I suspect this code path will only be reachable after we've removed any lone sstables.